BUILDER_DEV.md

```diff
@@ -1,390 +1,294 @@
+## Officially Supported Dev Environment
 
-Create the following files somewhere on your local filesystem:
+This Dev Environment involves spinning up a virtual machine (most of the core contributors use VMWare, but Virtual Box should work as well) running Ubuntu 17.10 desktop.
 
-`config_api.toml`
-```toml
-[depot]
-path = "/hab/svc/builder-api/data"
-key_dir = "/path/to/bldr-key-pair"
+### Pre-Reqs
+
+* Ubuntu 17.10 Desktop Virtual Machine
+* Access to the "core" origin in [Production Builder](https://bldr.habitat.sh). Ask a core maintainer (or in the [Habitat Slack](http://slack.habitat.sh/)) if you do not have access.
+* Access to the Habitat 1Password account (NOTE - we are working on a solution to make this not required, but it is necessary for the time being)
+* All of the following steps should be run within your Virtual Machine
+* A [Github Personal Access token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/) with all repo and all user permissions.
+* The [Habitat Builder App](https://github.com/apps/habitat-builder) installed on your Github account.
+* Open your Sudoers file at /etc/sudoers and remove the "secure_path" section (this is required for the make scripts to run correctly)
+
+```
+$ sudo visudo -f /etc/sudoers
+```
+
+### Builder Setup
+* Update your system
+```
+$ sudo apt-get update
+```
+
+* Install git
+```
+$ sudo apt-get install git
+```
+
+* Install curl
+```
+$ sudo apt-get install curl
+```
+
+* Clone the [Habitat repo](https://github.com/habitat-sh/habitat)
+```
+$ git clone https://github.com/habitat-sh/habitat.git
+```
+
+* Install the [Habitat CLI](https://www.habitat.sh/docs/get-habitat/)
+* Copy habitat-builder-dev.2017-10-02.private-key.pem from 1Password into path/to/habitat/repo/.secrets/builder-github-app.pem (NOTE - this is required as of 12/11/17 - we are working on a solution to make access to the Habitat 1Password vault not required for dev environment setup).
+* Set up some environmental variables (I put these in my .bashrc)
+
+```
+export HAB_AUTH_TOKEN="<your github token>"
+export HAB_BLDR_URL="http://localhost:9636"
+export HAB_ORIGIN="core"
+```
+
+* symlink /src to path/to/your/habitat/repo
+
+```
+$ sudo ln -s path/to/your/habitat/repo /src
 ```
 
-`config_jobsrv.toml`
-```toml
-key_dir = "/path/to/bldr-key-pair"
+* Cd into your Habitat repo
 
-[archive]
-backend = "local"
-local_dir = "/tmp"
 ```
+$ cd path/to/your/habitat/repo
+```
+
+* Run the provision script
+
+```
+$ sudo ./support/linux/provision.sh
+```
+
+* After the script completes, source cargo/rust
+```
+$ source $HOME/.cargo/env
+```
+
+* Run the build-srv makefile (NOTE - if you receive an error about cargo not being found, make sure you have removed the "secure_path" section of /etc/sudoers)
 
-`config_sessionsrv.toml`
-```toml
-[permissions]
-admin_team = 1995301
-build_worker_teams = [2555389]
-early_access_teams = [1995301]
 ```
+$ sudo make build-srv
```

Comment:
I don't think this command should need `sudo`. I initially ran into this error:
```
jbauman@ubuntu:~/habitat$ make build-srv
sh -c 'cd components/builder-api && cargo build'
error: failed to load source for a dependency on `urlencoded`

Caused by:
  Unable to update https://github.com/iron/urlencoded#8157212e

Caused by:
  failed to open: /home/jbauman/.cargo/git/.cargo-lock-git

Caused by:
  Permission denied (os error 13)
Makefile:230: recipe for target 'build-builder-api' failed
make: *** [build-builder-api] Error 101
```

But that was just because the `~/.cargo` dir was owned by root, and I don't think we want that. It can be fixed by running 

`sudo chown -R jbauman:jbauman /home/jbauman/.cargo/`
or more generalized
`sudo USER=$(id -un) GROUP=$(id -gn) chown -R $USER:$GROUP /home/jbauman/.cargo/`

But I'd prefer to see if I can figure out a good way to have cargo installed by the user rather than root. That will probably involve changing the [`install_dev_9_linux.sh`](https://github.com/habitat-sh/habitat/blob/master/support/linux/install_dev_9_linux.sh) script that `provision.sh` calls.

Reply:
On second thought, I think it would be better to run `provision.sh` without `sudo`. `install_dev_9_linux.sh` doesn't presume `sudo` since it adds `sudo` only to certain commands. I'm going to see if I can do the same for `provision.sh` so we're only using `sudo` where we really need it.